transport/rawsocketpeer.go

```diff
@@ -45,31 +45,55 @@ const (
 	magic = 0x7f
 )
 
-// ConnectRawSocketPeer creates a new rawSocketPeer with the specified config,
+// ConnectRawSocketPeer calls ConnectRawSocketPeerTimeout without a Dial timeout
+func ConnectRawSocketPeer(network, address string, serialization serialize.Serialization, logger stdlog.StdLog, recvLimit int) (wamp.Peer, error) {
+	return ConnectRawSocketPeerTimeout(0, network, address, serialization, logger, recvLimit)
+}
+
+// ConnectRawSocketPeerTimeout creates a new rawSocketPeer with the specified config,
 // and connects it to the WAMP router at the specified address.  The network
 // and address parameters are documented here:
 // https://golang.org/pkg/net/#Dial
 //
 // If recvLimit is > 0, then the client will not receive messages with size
 // larger than the nearest power of 2 greater than or equal to recvLimit.  If
 // recvLimit is <= 0, then the default of 16M is used.
-func ConnectRawSocketPeer(network, address string, serialization serialize.Serialization, logger stdlog.StdLog, recvLimit int) (wamp.Peer, error) {
-	err := checkNetworkType(network)
+func ConnectRawSocketPeerTimeout(
+	timeout time.Duration,
```

Comment:
For symmetry with websocket, please call this `ConnectRawSocketPeerContext` and pass `context.Context` as first arg.

Then, instead of using `net.DialTimeout` do:
```
var d net.Dialer
d.DialContext(ctx,...)
```